Dockerfile

```diff
@@ -4,102 +4,152 @@
 # them first, right off the bat lets us cache that and having node.js level
 # dependency changes not trigger a reinstall.
 RUN set -x \
     && apt-get update \
     && apt-get install --no-install-recommends -y \
-        libjpeg62 \
-    && apt-get autoremove -y \
-    && apt-get clean \
-    && rm -rf /var/lib/apt/lists/* /tmp/* /var/tmp/*
+        libjpeg-dev
 
 # However, we do want to trigger a reinstall of our node.js dependencies anytime
 # our package.json changes, so we'll ensure that we're copying that into our
 # static container prior to actually installing the npm dependencies.
-COPY package.json .babelrc /app/
+COPY package.json .babelrc /opt/warehouse/src/
 
 # Installing npm dependencies is done as a distinct step and *prior* to copying
 # over our static files so that, you guessed it, we don't invalidate the cache
 # of installed dependencies just because files have been modified.
 RUN set -x \
-    && apt-get update \
-    && apt-get install --no-install-recommends -y \
-        libjpeg-dev \
     && npm install -g gulp-cli \
-    && npm install \
-    && apt-get remove --purge -y libjpeg-dev \
-    && apt-get autoremove -y \
-    && apt-get clean \
-    && rm -rf /var/lib/apt/lists/* /tmp/* /var/tmp/*
+    && npm install
 
 # Actually copy over our static files, we only copy over the static files to
 # save a small amount of space in our image and because we don't need them. We
 # copy Gulpfile.babel.js last even though it's least likely to change, because
 # it's very small so copying it needlessly isn't a big deal but it will save a
 # small amount of copying when only Gulpfile.babel.js is modified.
-COPY warehouse/static/ /app/warehouse/static/
-COPY Gulpfile.babel.js /app/
+COPY warehouse/static/ /opt/warehouse/src/warehouse/static/
+COPY Gulpfile.babel.js /opt/warehouse/src/
 
 RUN gulp dist
 
 
-# Now we're going to build our actual application image, which will eventually
-# pull in the static files that were built above.
-FROM python:3.6.2-alpine3.6
 
-# Setup some basic environment variables that are ~never going to change.
-ENV PYTHONUNBUFFERED 1
-ENV PYTHONPATH /app/
 
-WORKDIR /app/
+
+
+# Now we're going to build our actual application, but not the actual production
+# image that it gets deployed into.
+FROM python:3.6.3-slim-stretch as build
 
 # Define whether we're building a production or a development image. This will
 # generally be used to control whether or not we install our development and
 # test dependencies.
 ARG DEVEL=no
 
-# Install System level Warehouse requirements, this is done before everything
-# else because these are rarely ever going to change.
+# This is a work around because otherwise libpq-dev bombs out trying
+# to create symlinks to these directories.
 RUN set -x \
-    && apk --no-cache add libpq libxml2 libxslt \
-        $(if [ "$DEVEL" = "yes" ]; then echo 'bash libjpeg postgresql-client'; fi)
+    && mkdir -p /usr/share/man/man1 \
+    && mkdir -p /usr/share/man/man7
```

Comment:
I think this can be removed? Pretty sure it's only `postgresql-client` that needs these to exist.